src/utils/parsers.rs

```diff
@@ -133,6 +133,19 @@ pub fn parse_url(url: &str) -> Result<Url> {
     Ok(res)
 }
 
+/// Parses a key:value pair from a string and returns a tuple of key:value.
+/// If a full key:value can't be parsed, it will error.
+pub fn parse_header(header: &str) -> Result<(String, String)> {
+    // only split once, a header's value may have a ":" in it, but not a key. Right?
```

Comment:
I think if it has a `:`, it'll likely be in a quoted string. The only `:` that's likely to be there is the one that separates the header key and the header value. The values are just going to be comma separated. That is to say the parsing you have below seems correct to me.

From the spec:
```console
message-header = field-name ":" [ field-value ]
field-name     = token
field-value    = *( field-content | LWS )
field-content  = <the OCTETs making up the field-value
                 and consisting of either *TEXT or combinations
                 of token, separators, and quoted-string>
```